### PR TITLE
Add Ex4

### PR DESCRIPTION
**Pull request by adifeiner**

### CODE REVIEW

File: Ex4.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     * The function check if the arguments legal and check
+     * all the options for tiles belong to the same island
+     * @param mapTiles - 2-dimensional array of booleans where false means water and true means land
+     * @param visit - 2-dimensional array of booleans
+     * @param numOfRows - number of rows
+     * @param numOfCols - number of columns
+     * @param rowIndex - row index in the tiles array
+     * @param colIndex - column index in te tiles array
+     */
+    private static void findAllTilesOfIsland(boolean[][] mapTiles, boolean[][] visit, int numOfRows, int numOfCols,
```
Comment (brzozan) [inline Ex4.java]: Minor naming nitpick: slightly more clear name could be something like `markAllTilesOfIsland`.

File: Ex4.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                    // find all the tiles that belong to this island
+                    findAllTilesOfIsland(mapTiles, visit, numOfRow, numOfCol, i, j);
+                }
+            }
+        }
+        return numOfIslands;
+    }
+
+    /**
+     * The function check if the arguments legal and check
```
Comment (brzozan) [inline Ex4.java]: Typically you do not need to write that method checks arguments. Just explain its main functionality, i.e. finding all tiles that belong to the island containing tile `(rowIndex, colIndex)` and marking them as visited.

File: Ex4Test.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class Ex4Test {
+
+    @Test
+    public void testCountIslands(){
+        boolean[][] mapTiles = new boolean[4][4];
```
Comment (brzozan) [inline Ex4Test.java]: You could also initialize array via literal shaped in a nice rectangle, which would make it more clear what the map looks like:

```
boolean[][] mapTiles = {
    { false, true,  false, false },
    { true,  true,  false, false },
...
};
```

But feel free to leave it as is.

File: Ex4.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+public class Ex4 {
+
+    /**
+     * The fuction loop over the mapTiles array and count the number of islands
+     * @param numOfRow
```
Comment (brzozan) [inline Ex4.java]: Please add comments to all parameters and return value.

Also minor stylistic comment: it's good to add empty line between main description of the method and the block of `@param` lines.

File: Ex4Test.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    @Test
+    public void testCountIslands(){
+        boolean[][] mapTiles = new boolean[4][4];
+        mapTiles[0][1] = true;
+        mapTiles[1][0] = true;
+        mapTiles[1][1] = true;
+        mapTiles[0][3] = true;
+        mapTiles[2][2] = true;
+        mapTiles[3][2] = true;
+        Ex4 countIslands = new Ex4();
```
Comment (brzozan) [inline Ex4Test.java]: All the methods are static, so no need to create instance of Ex4. Same in other tests.

**Review by brzozan — CHANGES_REQUESTED** [review]
Very good!

I had only cosmetic comments.

**Comment by adifeiner** [comment]
Updating with the fix code, thanks!

**Review by brzozan — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**